chemprop/cli/hpopt.py

```diff
@@ -315,11 +315,17 @@ def train_model(config, args, train_dset, val_dset, logger, output_transform, in
     model = build_model(args, train_loader.dataset, output_transform, input_transforms)
     logger.info(model)
 
-    monitor_mode = "min" if model.metrics[0].minimize else "max"
-    logger.debug(f"Evaluation metric: '{model.metrics[0].alias}', mode: '{monitor_mode}'")
+    if args.tracking_metric == "val_loss":
+        tracking_metric_class = model.criterion
+    else:
+        tracking_metric_class = MetricRegistry[args.tracking_metric]
+        args.tracking_metric = "val/" + args.tracking_metric
```

Comment:
I'm more in favor of adding this check and then giving a `logger.warn` message if the metric/loss function doesn't match the predictor. I think this would be easiest to add after merging in the change to torchmetrics PR.